### ISSUE TITLE
Remove ansible.cfg update requirements

### ISSUE THREAD

**Issue by fultonj** [fultonj/tripleo-ceph-ansible#15]
init.sh updates [1] ansible.cfg in the following ways but we cannot rely on that so this issue tracks removing these changes. 

1. Disabling Ansible host key checking
```
    sudo sed -i -e s/\#host_key_checking\ =\ False/host_key_checking=False/g /etc/ansible/ansible.cfg
```
2. Updating /etc/ansible/ansible.cfg action_plugins=/usr/share/ceph-ansible/plugins
```
    sudo sed -i -e s/\#action_plugins.*/action_plugins\ \=\ \\/usr\\/share\\/ceph-ansible\\/plugins\\/actions/g /etc/ansible/ansible.cfg
```

3. Disable retry files given permissions issue with /usr/share
```
    sudo sed -i s/\#retry_files_enabled\ =\ False/retry_files_enabled\ =\ False/g /etc/ansible/ansible.cfg
```

4. Disable deprecation warnings
```
    sudo sed -i s/\#deprecation_warnings\ =\ True/deprecation_warnings\ =\ False/g /etc/ansible/ansible.cfg
```

[1] https://github.com/fultonj/tripleo-ceph-ansible/blob/b4d01a2f81b104e6c4a93b4bd04c4200a2cc811c/init.sh#L48-L59

**Comment by fultonj** [comment]
Plan:
1. This should be fixed by adding `--ssh-extra-args='-o StrictHostKeyChecking=no'` to https://review.openstack.org/#/c/469644/13/workbooks/ceph-ansible.yaml 
2. Reasons for this are not clear; reproduce the issue and debug with seb 
3. File an RFE with the ceph-ansible project (as per seb)
4. No change; if deprecation warnings are left it doesn't affect the playbook run so it was just a nice-to-have

**Comment by fultonj** [comment]
Regarding 3, even if [ansible.cfg](https://github.com/ceph/ceph-ansible/blob/master/ansible.cfg), as provided by the ceph-ansible package which ends up in `/usr/share/ceph-ansible/ansible.cfg`, could be modified, the user would still need to do the modification.

**Comment by fultonj** [comment]
For item 3 I am now waiting on the ceph-ansible project to respond to my request: https://github.com/ceph/ceph-ansible/issues/1611

**Comment by fultonj** [comment]
For item 1, I am debugging. I undid what #1 does:
```
sudo crudini --set /etc/ansible/ansible.cfg defaults host_key_checking True 
```
I added the following to ceph-ansible.yaml
```
            ssh_extra_args: '-o StrictHostKeyChecking=no'
```
I see my playbook was called with the new option but I am getting an SSH key warning: 
```
Unexpected error while running command.
Command: ansible-playbook /usr/share/ceph-ansible/site-docker.yml.sample --user heat-admin --become --become-user root --extra-vars {"ceph_mon_docker_subnet": "192.168.24.0/24", "monitor_secret": "***", "ceph_rgw_civetweb_port": 8080, "cluster": "ceph", "fetch_directory": "/tmp/ceph-ansible-fetch/", "ceph_docker_on_openstack": false, "rgw_containerized_deployment": true, "ceph_docker_image_tag": "tag-build-master-jewel-centos-7", "osd_containerized_deployment": true, "ceph_osd_docker_devices": ["/dev/vdb", "/dev/vdc"], "public_network": "192.168.24.0/24", "journal_collocation": true, "ssh_extra_args": "-o StrictHostKeyChecking=no", "mds_containerized_deployment": true, "mon_containerized_deployment": true, "cluster_network": "192.168.24.0/24", "fsid": "b007365e-cc87-11e5-a877-525400330ee0", "journal_size": 256, "ceph_mon_docker_interface": "br_ex", "openstack_config": true, "devices": ["/dev/vdb", "/dev/vdc"], "docker": true} --forks 6 --inventory-file /tmp/ansible-mistral-action6BKjrB/inventory.yaml --private-key /tmp/ansible-mistral-action6BKjrB/ssh_private_key
Exit code: 4
Stdout: u'\
PLAY [mons] ********************************************************************\
\
TASK [setup] *******************************************************************\
fatal: [192.168.24.7]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\\\\r\\\
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @\\\\r\\\
```
Reviewing other available options from:

https://github.com/openstack/tripleo-common/commit/e6c8a46f00436edfa5de92e97c3a390d90c3ce54

**Comment by fultonj** [comment]
I had put `ssh_extra_args: '-o StrictHostKeyChecking=no'` under extra_vars not input originally. Working now. 

So 1 and 4 are resolved. That leaves two changes:

2. retry files (https://github.com/ceph/ceph-ansible/issues/1611)
3. action_plugins

**Comment by fultonj** [comment]
Regarding item 3 (action_plugins). If I don't do the following to the default ansible.cfg shipped with TripleO,
```
crudini --set /etc/ansible/ansible.cfg defaults action_plugins /usr/share/ceph-ansible/plugins/actions
```
Then I see the following error:
```
TASK [ceph-mon : generate ceph.conf configuration file] ************************\
fatal: [192.168.24.11]: FAILED! => {"failed": true, "msg": "module (config_template) is missing interpreter line"}\
\
PLAY RECAP *********************************************************************\
192.168.24.11              : ok=21   changed=4    unreachable=0    failed=1   \
192.168.24.17              : ok=1    changed=0    unreachable=0    failed=0   \
192.168.24.8               : ok=1    changed=0    unreachable=0    failed=0   \
```
Full error log at http://sprunge.us/JODV

**Comment by fultonj** [comment]
Can the ansible environment be used for this? https://docs.ansible.com/ansible/playbooks_environment.html

**Comment by fultonj** [comment]
Update: 
2. retry files (ceph/ceph-ansible#1611) The ansible.cfg that ceph-ansible ships will now have retry files disabled by default 
3. action_plugins The action plugs are referenced in the ansible.cfg that ceph-ansible ships, if only I could call it

So this whole issue can be resolved by getting Mistral to call ansible-playbook with the ansible.cfg that it ships.

**Comment by fultonj** [comment]
Q: How does ansible-playbook know which ansible.cfg to use? 
A: Based on the directory it is running in (if none is there it uses the one in /etc/) OR the env var: ANSIBLE_CONFIG

In /home/stack
```
(undercloud) [stack@undercloud ~]$ ansible-playbook --version | grep config 
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
(undercloud) [stack@undercloud ~]$ 
```

In /usr/share/ceph-ansible
```
(undercloud) [stack@undercloud ceph-ansible]$ ansible-playbook --version | grep config 
[WARNING]: log file at /var/log/ansible.log is not writeable and we cannot create it, aborting

  config file = /usr/share/ceph-ansible/ansible.cfg
  configured module search path = Default w/o overrides
(undercloud) [stack@undercloud ceph-ansible]$ 
```
Or wherever, provided you set the env variable
```
(undercloud) [stack@undercloud ~]$ ANSIBLE_CONFIG=/usr/share/ceph-ansible/ansible.cfg ansible-playbook --version
[WARNING]: log file at /var/log/ansible.log is not writeable and we cannot create it, aborting

ansible-playbook 2.2.1.0
  config file = /usr/share/ceph-ansible/ansible.cfg
  configured module search path = Default w/o overrides
(undercloud) [stack@undercloud ~]$ 
```

**Comment by fultonj** [comment]
Thus, I need to submit a review to make [tripleo-common's ansible.py](https://review.openstack.org/#/c/473587/6/tripleo_common/actions/ansible.py) set a `env_variables` not just to HOME but to something I can pass it as extra environment variables. Then I should be able to pass an extra dictionary and make it contain `ANSIBLE_CONFIG=/usr/share/ceph-ansible/ansible.cfg`. 

Next step: get Juan to restore https://review.openstack.org/#/c/475728/2

**Comment by fultonj** [comment]
flapper87 to the rescue with an update of it too. Next step, test it. 
 https://review.openstack.org/#/c/475728/

**Comment by fultonj** [comment]
Tested https://review.openstack.org/#/c/475728/ and set verbosity: 5. Observed the following with the output of see_last_task.sh: 
```
  "stderr": "Using /usr/share/ceph-ansible/ansible.cfg as config file
```
Playbook run succeeded with `#action_plugins = /usr/share/ceph-ansible/plugins/actions` in /etc/ansible/ansible.cfg and with disable retries only in ceph-ansbile's ansible.cfg. 

```
(undercloud) [stack@undercloud tripleo-ceph-ansible]$ grep retry /etc/ansible/ansible.cfg | grep -v \#
retry_files_enabled = True
(undercloud) [stack@undercloud tripleo-ceph-ansible]$
(undercloud) [stack@undercloud tripleo-ceph-ansible]$ grep retry /usr/share/ceph-ansible/ansible.cfg  | grep -v \#
retry_files_enabled = False
(undercloud) [stack@undercloud tripleo-ceph-ansible]$ 
```
